extensions/twingate/src/recent-resources.tsx

```diff
@@ -0,0 +1,227 @@
+import {
+  ActionPanel,
+  Action,
+  List,
+  showToast,
+  Toast,
+  Icon,
+  getPreferenceValues,
+} from "@raycast/api";
+import { useCachedPromise } from "@raycast/utils";
+import React, { useState, useEffect } from "react";
+import { TwingateStorage } from "./utils/storage";
+import { RecentResource } from "./types";
+import { DebugLogger } from "./utils/debug";
+import { ResourceListItem } from "./components/ResourceListItem";
+import { useFavorites } from "./hooks/useFavorites";
+import { parseShortcut, DEFAULT_SHORTCUTS } from "./utils/shortcuts";
+
+interface Preferences {
+  apiKey: string;
+  networkUrl: string;
+  shortcutRefreshResources: string;
+  shortcutClearData: string;
+  shortcutDebugMode: string;
+  shortcutExportLogs: string;
+  shortcutToggleFavorite: string;
+  shortcutCopyUrl: string;
+  shortcutCopyAddress: string;
+  shortcutCopyAlias: string;
+  shortcutCopyName: string;
+  shortcutOpenMainSearch: string;
+}
```

Comment:
You don't need to manually set preference types as this is autogenerated in `raycast-env.d.ts`